Scripts/Editor/StatisticWindow.cs

```diff
@@ -0,0 +1,791 @@
+                    {
+                        _statCache[index].frameIndex = (int)((Event.current.mousePosition.x - r.x) / r.width * _dataCache[fileIndex].frameCount);
+                        Repaint();
+                    }
+                    GUI.DrawTexture(r, _statCache[index].graph);
+                    Rect r2 = new Rect(r);
+                    r2.x = r.x + (((float)_statCache[index].frameIndex / (float)_dataCache[fileIndex].frameCount) * r.width);
+                    r2.width = 2;
+                    GUI.DrawTexture(r2, _indicator);
+                    GUI.Label(r, _statCache[index].positive.ToString(), Styles.textTopLeft);
+                    GUI.Label(r, _statCache[index].negative.ToString(), Styles.textBottomLeft);
+                }
+            }
+            EditorGUILayout.EndVertical();
+        }
+
+        private int GetStatIndex(RecordMessage message, int desiredIndex)
+        {
+            int v = 0;
+            for (int i = 0; i < message.frames.Count; i++)
+            {
+                if(message.frames[i] <= desiredIndex)
+                {
+                    v = i;
+                }
+            }
+            return v;
+        }
+
+        private string SingleStatBox(object item)
+        {
+            string output = "";
+            if (item.GetType() == typeof(string) ||
+                item.GetType() == typeof(int) ||
+                item.GetType() == typeof(float) ||
+                item.GetType() == typeof(double))
+            {
+                output = EditorGUILayout.TextField(item.ToString());
+                CopyButton(output);
+            }
+            else if(item.GetType() == typeof(Vector2))
+            {
+                Vector2 item2 = (Vector2)item;
+                output = EditorGUILayout.TextField("("+ item2.x.ToString()+", "+item2.y.ToString()+")");
+                CopyButton(output);
+            }
+            else if(item.GetType() == typeof(Vector3))
+            {
+                Vector3 item3 = (Vector3)item;
+                output = EditorGUILayout.TextField("(" + item3.x.ToString() + ", " + item3.y.ToString() + ", " + item3.z.ToString() + ")");
+                CopyButton(output);
+            }
+            else if (item.GetType() == typeof(Vector4))
+            {
+                Vector4 item4 = (Vector4)item;
+                output = EditorGUILayout.TextField("(" + item4.x.ToString() + ", " + item4.y.ToString() + ", " + item4.z.ToString() + ", " + item4.w.ToString() + ")");
+                CopyButton(output);
+            }
+            return output;
+        }
+
+        private void CopyButton(string toCopy)
+        {
+            if(GUILayout.Button(_copyButton,GUILayout.Width(Sizes.widthCharButton)))
+            {
+                GUIUtility.systemCopyBuffer = toCopy;
+            }
+        }
+    }
+
```

Comment:
This is a huge class (so much so that github refuses to automatically load it), would be good to break it up a bit more